src/webgpu/api/operation/buffers/map_oom.spec.ts

```diff
@@ -125,15 +126,29 @@ an out-of-memory error if allocation fails.
       .combine('usage', kBufferUsages)
   )
   .fn(async t => {
-    const { usage, size } = t.params;
+    const { oom, usage, size } = t.params;
 
-    const buffer = t.expectGPUError('out-of-memory', () =>
-      t.device.createBuffer({ mappedAtCreation: true, size, usage })
+    const buffer = t.expectGPUError(
+      'out-of-memory',
+      () => t.device.createBuffer({ mappedAtCreation: true, size, usage }),
+      oom
     );
 
-    // Smaller range inside a too-big mapping
-    const mapping = buffer.getMappedRange(0, 16);
-    t.expect(mapping.byteLength === 16);
-    buffer.unmap();
-    t.expect(mapping.byteLength === 0, 'Mapping should be detached');
+    const f = () => buffer.getMappedRange(0, 16);
+
+    let mapping: ArrayBuffer | undefined = undefined;
+    if (oom) {
+      // Smaller range inside a too-big mapping
+      t.shouldThrow('OperationError', () => {
+        mapping = f();
```

Comment:
Shouldn't getMappedRange here succeed, and return a dummy allocation?

I know the content process _can_ already know that there was a shmem OOM, but IIRC we don't want to expose the difference between shmem OOM and vram OOM (cc @austinEng in case I'm forgetting something again, this is really hard to hold in my head ^_^).

Unfortunately we can't easily trigger vram OOM without triggering shmem OOM. Actually... we should probably have a stress test that allocates as much vram as it can, in large chunks, until it reaches vram-OOM, then does this test. I have some ideas about this so I'll put notes in src/stress/README.txt.

Reply:
ah, yea it should attempt to allocate dummy memory. It would only throw an exception if `getMappedRange` was too big. I missed this when reviewing the Dawn patch.

we should make sure this test is right, but we can probably defer creating a dummy allocation until post-OT. GIANT mappedAtCreation + getMappedRange is probably a bit niche. I think this is the only scenario where it's needed?

Reply:
Fixed the test first. Worry about the chrome impl later (Thanks for filing: https://bugs.chromium.org/p/dawn/issues/detail?id=995)